### PR TITLE
feat: Add retry to API requests

### PR DESCRIPTION
**Pull request by erezrokah**
Looks like this is the only way to inject the retry client, there's also https://github.com/deepmap/oapi-codegen/issues/150 which is in progress via https://github.com/deepmap/oapi-codegen/pull/156

### CODE REVIEW

File: client.gen.go [hunk 1]
```diff
@@ -48,6 +49,7 @@ type ClientOption func(*Client) error
 
 // Creates a new Client, with reasonable defaults
 func NewClient(server string, opts ...ClientOption) (*Client, error) {
```
Comment (erezrokah) [inline client.gen.go:51]: There's also `NewClientWithResponses` but it wraps the call to `NewClient`

File: client.gen.go [hunk 2]
```diff
@@ -48,6 +49,7 @@ type ClientOption func(*Client) error
...
+	opts = append([]ClientOption{WithHTTPClient(retryablehttp.NewClient().StandardClient())}, opts...)
```
Comment (erezrokah) [inline client.gen.go:52]: I'm prepending the option to allow consumers override it

**Review by hermanschaaf — APPROVED** [review]
Bit complicated 🙃 ... hopefully we can find a simpler way to do this in the future

**Comment by erezrokah** [comment]
> Bit complicated 🙃 ... hopefully we can find a simpler way to do this in the future

Yeah I could add `NewClientWithRetries` as well, but it requires updating all consumers

**Pull request merged**